addons/hr/models/models.py

```diff
@@ -19,3 +21,72 @@ def _alias_get_error(self, message, message_dict, alias):
                 return AliasError('error_hr_employee_restricted', _('restricted to employees'))
             return False
         return super()._alias_get_error(message, message_dict, alias)
+
+    def _map_domain(self, domain, map_function: callable) -> Domain:
+        """
+        Applies map_function to each condition in the domain.
+
+        :param domain: The original domain to process. Remains unmodified.
+        :param map_function: A callable that takes a condition and returns either:
+            - A transformed condition, or
+            - None (discards the condition).
+        :return: A new domain containing only the transformed conditions.
+        """
+        domain = Domain(domain)
+        stack = []
+        for condition in list(domain)[::-1]:
+            if condition in ['&', '|']:
+                operator = condition
+                cond_a = stack.pop()
+                cond_b = stack.pop()
+                if cond_a and cond_b:
+                    stack.append([operator, cond_a, cond_b])
+                elif cond_a:
+                    stack.append(cond_a)
+                elif cond_b:
+                    stack.append(cond_b)
+                else:
+                    stack.append(None)
+            else:
+                stack.append(map_function(condition))
+
+        def _flatten_stack(stack: list) -> list:
```

Comment:
Nitpick. You could write it as: first map your function, then compact to remove None.